app/processing/get-completed-payment-requests-filter.js

```diff
@@ -0,0 +1,34 @@
+const db = require('../data')
+const { BPS, CS } = require('../constants/schemes')
+
+const getCompletedPaymentRequestsFilter = (paymentRequest) => {
+  switch (paymentRequest.schemeId) {
+    case BPS:
+      return {
+        schemeId: paymentRequest.schemeId,
+        frn: paymentRequest.frn,
+        marketingYear: paymentRequest.marketingYear
+      }
+    case CS:
+      return {
+        schemeId: paymentRequest.schemeId,
+        frn: paymentRequest.frn,
+        marketingYear: paymentRequest.marketingYear,
+        [db.Sequelize.Op.or]: [
```

Comment:
I think I sort of understand what thing means, but just wanted to ask what it does any way

Reply:
We're using Sequelize as an ORM so what it's doing is it's converting this JavaScript into SQL code that it will run against the database.

Basically what we're building here is the `WHERE` clause of a `SELECT` query.

So where we're just retuning the values, eg `schemeId: paymentRequest.schemeId` that will get translated to

`WHERE "schemeId" = 1` (if payment request scheme id was 1)

But with CS we need to us an `OR` in the `WHERE` so Sequelize has built in operators to do that.  Personally I find the syntax a bit more complicated than it should be as it uses symbols.  But `db.Sequelize.Op.or` translates to `OR` 

Then you just pass in the alternatives in an array.  So it would become something like this

`WHERE "contractNumber" = 'A0123456' OR "contractNumber = 'A123456'`

Reply:
Yeah, that's what I gathered from it, just wasn't sure - thanks :)